lib/k8s/stack.rb

```diff
@@ -101,6 +101,9 @@ def apply(client, prune: true)
         if !server_resource
           logger.info "Create resource #{resource.apiVersion}:#{resource.kind}/#{resource.metadata.name} in namespace #{resource.metadata.namespace} with checksum=#{resource.checksum}"
           keep_resource! client.create_resource(prepare_resource(resource))
+        elsif server_resource.metadata&.ownerReferences && !server_resource.metadata.ownerReferences.empty?
+          logger.info "Server resource #{server_resource.apiVersion}:#{server_resource.apiKind}/#{server_resource.metadata.name} in namespace #{server_resource.metadata.namespace} has an ownerReference and will be kept"
+          keep_resource! server_resource
```

Comment:
hmm, I don't think this will work.
```
keep_resource! server_resource
```
will make `@keep_resources` map to have `nil` value for the "parsed" resource key. It'll later on fail at `keep_resource?` on `return false unless keep_annotation` as the `keep_annotation` will be `nil`